auth/pat.go

```diff
@@ -95,592 +95,188 @@ func ParseOperationType(ot string) (OperationType, error) {
-	if s == nil {
-		return nil
-	}
-	switch platformEntityType {
-	case PlatformUsersScope:
-		if err := s.Users.Delete(operation, entityIDs...); err != nil {
-			return fmt.Errorf("failed to delete platform %s scope: %w", platformEntityType.String(), err)
-		}
-	case PlatformDashBoardScope:
-		if err := s.Dashboard.Delete(operation, entityIDs...); err != nil {
-			return fmt.Errorf("failed to delete platform %s scope: %w", platformEntityType.String(), err)
-		}
-	case PlatformMesagingScope:
-		if err := s.Messaging.Delete(operation, entityIDs...); err != nil {
-			return fmt.Errorf("failed to delete platform %s scope: %w", platformEntityType.String(), err)
-		}
-	case PlatformDomainsScope:
-		if optionalDomainID == "" {
-			return fmt.Errorf("failed to delete platform %s scope: invalid domain id", platformEntityType.String())
-		}
-		ds, ok := s.Domains[optionalDomainID]
-		if !ok {
-			return nil
-		}
-		if err := ds.Delete(optionalDomainEntityType, operation, entityIDs...); err != nil {
-			return fmt.Errorf("failed to delete platform %s id %s  scope : %w", platformEntityType.String(), optionalDomainID, err)
-		}
-	default:
-		return fmt.Errorf("failed to add platform %d scope: invalid platform entity type ", platformEntityType)
+	if optionalDomainID != "" && s.OptionalDomainId != optionalDomainID {
+		return false
 	}
-	return nil
-}
 
-// Check entry in Domain scope.
-func (s *Scope) Check(platformEntityType PlatformEntityType, optionalDomainID string, optionalDomainEntityType DomainEntityType, operation OperationType, entityIDs ...string) bool {
-	if s == nil {
+	if s.Operation != operation {
 		return false
 	}
-	switch platformEntityType {
-	case PlatformUsersScope:
-		return s.Users.Check(operation, entityIDs...)
-	case PlatformDashBoardScope:
-		return s.Dashboard.Check(operation, entityIDs...)
-	case PlatformMesagingScope:
-		return s.Messaging.Check(operation, entityIDs...)
-	case PlatformDomainsScope:
-		ds, ok := s.Domains[optionalDomainID]
-		if !ok {
-			return false
-		}
-		return ds.Check(optionalDomainEntityType, operation, entityIDs...)
-	default:
-		return false
+
+	if s.EntityId == "*" {
+		return true
 	}
-}
 
-func (s *Scope) String() string {
-	str, err := json.Marshal(s) // , "", "  ")
-	if err != nil {
-		return fmt.Sprintf("failed to convert scope to string: json marshal error :%s", err.Error())
+	if s.EntityId == entityID {
+		return true
 	}
-	return string(str)
+	return false
 }
 
 // PAT represents Personal Access Token.
 type PAT struct {
 	ID          string    `json:"id,omitempty"`
-	User        string    `json:"user,omitempty"`
+	User        string    `json:"user_id,omitempty"`
 	Name        string    `json:"name,omitempty"`
 	Description string    `json:"description,omitempty"`
 	Secret      string    `json:"secret,omitempty"`
-	Scope       Scope     `json:"scope,omitempty"`
+	Scope       []Scope   `json:"scope,omitempty"`
```

Comment:
I think we should not have `scopes` in `PAT`
Because it have problem if there are 1000 of scopes 

If we move scope out of PAT, then this requires additional endpoint with pagination to retrieve scopes